hammers/scripts/gpu_lease_stacking.py

```diff
@@ -0,0 +1,145 @@
+# coding: utf-8
+'''
+.. code-block:: bash
+
+    gpu-lease-stacking {info, delete}
+
+Reclaims GPU nodes from leases that violate terms of use.
+
+* ``info`` to just display leases or actuall delete them with ``delete``
+'''
+import sys
+import argparse
+from pprint import pprint
+from datetime import datetime
+
+from hammers import MySqlArgs, osapi, query
+from hammers.slack import Slackbot
+from hammers.osrest.blazar import lease_delete
+
+
+def find_stacked_leases(leases):
+    """Return list of only the leases stacked on each other."""
+    stacked = []
+
+    for i in range(len(leases)):
+
+        start_date = leases[i][1]
+        end_date = leases[i][2]
+
+        if i > 0:
+            last_end_date = leases[i - 1][2]
+        else:
+            last_end_date = datetime.min
+
+        if i < len(leases) - 1:
+            next_start_date = leases[i + 1][1]
+        else:
+            next_start_date = datetime.max
+
+        stacked_previous = (start_date - last_end_date).days < 1
+        stacked_next = (next_start_date - end_date).days < 1
+
+        if stacked_previous or stacked_next:
+            stacked.append(leases[i])
+
+    return stacked
+
+
+def reaper(db, auth, describe=False, quiet=False):
+    """Delete stacked leases on gpu nodes."""
+    user_gpu_leases = {}
+
+    for row in query.gpu_leases(db):
+        user_id = row['user_id']
+        node_id = row['node_id']
+
+        if user_id not in user_gpu_leases.keys():
```

Comment:
I don't think you need to use `.keys()` here?

```suggestion
        if user_id not in user_gpu_leases:
```